pytorch_lightning/overrides/data_parallel.py

```diff
@@ -285,7 +285,7 @@ def _worker(i, module, input, kwargs, device=None):
                 if output is None:
                     warn_missing_output(fx_called)
 
-                if output is not None and (module.use_dp or module.use_ddp2):
+                if output is not None and module.distrib_type in ("dp", "ddp2"):
```

Comment:
Use enum here?

Reply:
this is a different story, this PR work only with `use_*` so lest eventually make this replacement in another PR